core/src/test/java/org/infinispan/xsite/irac/ManualIracManager.java

```diff
@@ -57,7 +55,7 @@ public synchronized void trackUpdatedKey(int segment, Object key, Object lockOwn
    }
 
    @Override
-   public void trackExpiredKey(int segment, Object key, Object lockOwner) {
+   public synchronized void trackExpiredKey(int segment, Object key, Object lockOwner) {
```

Comment:
I must admit seeing synchronized everywhere in this class invoking super methods is a bit scary at first. But it seems all of the super methods do not acquire other locks or do anything too long running. But I just wanted to point that out as something to pay attention to.

Reply:
I can remove the synchronized if I use a `ConcurrentHashMap` instead :+1:

Reply:
Sounds good 👍